newIDE/app/src/Profile/EditProfileDialog.js

```diff
@@ -27,115 +28,122 @@ type Props = {|
-    form: {
-      username: this.props.profile.username || '',
-      description: this.props.profile.description || '',
-      getGameStatsEmail: !!this.props.profile.getGameStatsEmail,
-    },
-  };
+const EditDialog = ({
+  profile,
+  onClose,
+  onEdit,
+  editInProgress,
+  error,
+}: Props) => {
+  const [username, setUsername] = React.useState(profile.username || '');
+  const [description, setDescription] = React.useState(
+    profile.description || ''
+  );
+  const [getGameStatsEmail, setGetGameStatsEmail] = React.useState(
+    !!profile.getGameStatsEmail
+  );
+  const [
+    usernameAvailability,
+    setUsernameAvailability,
+  ] = React.useState<?UsernameAvailability>(null);
+  const [
+    isValidatingUsername,
+    setIsValidatingUsername,
+  ] = React.useState<boolean>(false);
 
-  _onEdit = () => {
-    if (!this._canEdit()) return;
+  const canEdit = React.useMemo(
+    () =>
+      !editInProgress &&
+      isUsernameValid(username, { allowEmpty: false }) &&
+      !isValidatingUsername &&
+      (!usernameAvailability || usernameAvailability.isAvailable),
+    [editInProgress, username, isValidatingUsername, usernameAvailability]
+  );
 
-    const { form } = this.state;
-    this.props.onEdit(form);
+  const edit = () => {
+    if (!canEdit) return;
+    onEdit({
+      username,
+      description,
+      getGameStatsEmail,
+    });
   };
 
-  _canEdit = () => {
-    return (
-      !this.props.editInProgress && isUsernameValid(this.state.form.username)
-    );
-  };
+  const actions = [
+    <FlatButton
+      label={<Trans>Back</Trans>}
+      disabled={editInProgress}
+      key="back"
+      primary={false}
+      onClick={onClose}
+    />,
+    <LeftLoader isLoading={editInProgress || isValidatingUsername} key="edit">
+      <DialogPrimaryButton
+        label={<Trans>Save</Trans>}
+        primary
+        disabled={!canEdit}
+        onClick={edit}
+      />
+    </LeftLoader>,
+  ];
 
-  render() {
-    const { onClose, editInProgress, error } = this.props;
-    const actions = [
-      <FlatButton
-        label={<Trans>Back</Trans>}
-        disabled={editInProgress}
-        key="back"
-        primary={false}
-        onClick={onClose}
-      />,
-      <LeftLoader isLoading={editInProgress} key="edit">
-        <DialogPrimaryButton
-          label={<Trans>Save</Trans>}
-          primary
-          onClick={this._onEdit}
-          disabled={!this._canEdit()}
+  return (
+    <Dialog
+      title={<Trans>Edit your GDevelop profile</Trans>}
+      actions={actions}
+      maxWidth="sm"
+      cannotBeDismissed={editInProgress}
+      onRequestClose={onClose}
+      onApply={edit}
+      open
+    >
+      <ColumnStackLayout noMargin>
+        <UsernameField
+          initialUsername={profile.username}
+          value={username}
+          onChange={(e, value) => {
+            setUsername(value);
+          }}
+          errorText={getUsernameErrorText(error)}
+          onAvailabilityChecked={(
```

Comment:
`onAvailabilityChecked={setUsernameAvailability}` is not working? Same for the CreateAccountDialog